### ISSUE TITLE
Wrong state in the mail subject for array tasks

### ISSUE THREAD

**Issue by Aelmazaty** [neilmunday/slurm-mail#109]
**Versions**

OS version: Rocky 8.5
Slurm version: 23.02.7
Slurm Mail version: 4.4

**Describe the bug**

A clear and concise description of what the bug is and how it to produce it.
It works perfectly for individual jobs. However for arrays, and when using --mail-type=ARRAY_TASKS, mails are sent with random (And mostly wrong) state in the subject. Although the exit code and exit state included in the email show otherwise.
I was able to reproduce this by having some successfult tasks and some failed tasks in the same array

**Logs**

Include any log snippets here - please used code tags.

<img width="694" alt="image" src="https://github.com/neilmunday/slurm-mail/assets/14933846/82913a1e-2cc7-466a-af23-c601b97a9851">

`# sacct -j 48971497
JobID           JobName  Partition    Account  AllocCPUS      State ExitCode
------------ ---------- ---------- ---------- ---------- ---------- --------
48971486_2   test_codon   standard    systems          1  COMPLETED      0:0
48971486_2.+      batch               systems          1  COMPLETED      0:0
48971486_2.+     extern               systems          1  COMPLETED      0:0`

**Comment by neilmunday** [comment]
Thank you for the bug report. I will see if I can recreate it.

**Comment by neilmunday** [comment]
I've re-created the bug... working on a fix.

**Comment by neilmunday** [comment]
I've implemented a bug fix in [c38287e](https://github.com/neilmunday/slurm-mail/commit/c38287e681b740510b4e81c5b71b19c042b5d806).

If you'd like to try out the fix you can apply the changes to `/usr/lib/python3.6/site-packages/slurmmail`:

```
-                if "{0}".format(first_job_id) not in sacct_dict["JobId"]:
+                job_id = int(sacct_dict["JobIdRaw"])
+
+                if not array_summary and job_id != int(first_job_id):
+                    logging.debug("skipping %s, it does not equal %s", job_id, first_job_id)
+                    print("skipping %s, it does not equal %s" % (job_id, first_job_id))
+                    continue
+
+                if array_summary and "{0}".format(first_job_id) not in sacct_dict["JobId"]:
+                    logging.debug("skipping %s for job array summary", sacct_dict["JobId"])
                     continue

-                job_id = int(sacct_dict["JobIdRaw"])
```

Alternatively, you can wait for version 4.10 of Slurm-Mail to be released which should be due in the next few days.

**Comment by neilmunday** [comment]
Version 4.10 released which includes the fix.
